api/filtering/db_custom_filters.py

```diff
@@ -116,7 +116,7 @@ def separate_operating_system_filters(filter_param) -> list[OsComparison]:
         return [OsComparison(comparator=filter_param)]
 
     # filter_param is a dict
-    for os_name in filter_param.keys():
+    for os_name in filter_param.keys(): # this doesn't account for "os_name" instead of os names
         if os_name not in (os_names := _get_valid_os_names()):
```

Comment:
I'd recommend checking for the key "name" between these 2 lines and doing the custom logic there:
```suggestion
    for os_name in filter_param.keys():
        if os_name == "name":
            # Return an OsComparison object that implies all hosts with a matching OS name.
            # Maybe if OsComparison.comparator is "eq" or "neq" and the major version is None?
        elif os_name not in (os_names := _get_valid_os_names()):
```